tests/test_core.py

```diff
@@ -1411,6 +1431,8 @@ def test_exceptions_3(self):
     self.do_run(src, 'Caught exception: Hello\nDone.', ['2'], no_build=True)
 
   def test_exceptions_white_list(self):
+    if self.is_wasm_backend():
```

Comment:
IIUC The decorator short-circuits the `do_run_from_file` bits but the code here does other stuff as well, so the test still registers as FAIL with the decorator.